shell/browser/net/proxying_url_loader_factory.cc

```diff
@@ -465,15 +533,34 @@ void ProxyingURLLoaderFactory::InProgressRequest::
       current_response_->headers = override_headers_;
     }
   }
+
+  if (for_cors_preflight_ && !redirect_url_.is_empty()) {
+    OnRequestError(network::URLLoaderCompletionStatus(net::ERR_FAILED));
```

Comment:
Can you add a test to check for modifying headers with redirect url fails in `onBeforeSendHeaders` and `onHeadersReceived`.

Reply:
The `redirect_url_` can only be changed in `onBeforeRequest`, but once we do that this code path won't be run. It seems that this check is only to guard edge cases that do not usually happen.